src/helpers.js

```diff
@@ -117,25 +119,85 @@ export function isTraversable(tileID, me) {
   }
 }
 
+/* eslint no-param-reassign: ["error", { "ignorePropertyModificationsFor": ["bigGrid"] }] */
+
+/*
+ * place all values from smallGrid into bigGrid. Align the upper left corner at x, y
+ */
+function fillGridWithSubgrid(bigGrid, smallGrid, x, y) {
+  for (let i = 0; i < smallGrid.length; i++) {
+    for (let j = 0; j < smallGrid[0].length; j++) {
+      bigGrid[i + x][j + y] = smallGrid[i][j];
+    }
+  }
+}
+
+
+/* Returns a 2d cell array of traversible (1) and blocked (0) cells inside a tile.
+ *
+ * tileTraverse: if this tile is traversable
+ * cpt: number of cells per tile
+ * objRadius: radius of untraversable object in pixels
+ */
+export function traversableCellsInTile(tileTraverse, cpt, objRadius) {
+  const gridTile = [];
+  // Start with all traversable
+  let i;
+  let j;
+  for (i = 0; i < cpt; i++) {
+    gridTile[i] = new Array(cpt);
+    for (j = 0; j < cpt; j++) {
+      gridTile[i][j] = 1;
+    }
+  }
+
+  if (!tileTraverse) {
+    const midCell = (cpt - 1.0) / 2.0;
+    for (i = 0; i < cpt; i++) {
+      for (j = 0; j < cpt; j++) {
+        const xDiff = Math.max(Math.abs(i - midCell) - 0.5, 0);
+        const yDiff = Math.max(Math.abs(j - midCell) - 0.5, 0);
+        const cellDist = Math.sqrt((xDiff * xDiff) + (yDiff * yDiff));
+        const pixelDist = cellDist * (PIXEL_PER_TILE / cpt);
+        if (pixelDist <= objRadius) {
+          // This cell touches the object, is not traversable
+          gridTile[i][j] = 0;
+        }
+      }
+    }
+  }
+  return gridTile;
+}
+
+
+/* eslint-disable one-var, no-unused-vars*/ // TODO: remove this
 
 /*
- * Returns a 2D array of traversable (1) and blocked (0) tiles.
+ * Returns a 2D array of traversable (1) and blocked (0) tiles. Size of return grid is
```

Comment:
good catch. originally, I think the plan was to make it return a grid of cells. Now that I think about it, there's already uinittests for the `getTraversableTiles` function, so those should just be updated and pass `cpt` as 1.

When I come back to this, I will fix terminology. This function is also untested so I have no clue if it works.